R/filter_state_api_NEW.R

```diff
@@ -0,0 +1,442 @@
+            slices[[length(slices)+1]] <- as.teal_slice(Filter(Negate(is.null), args))
+            args <- make_args()
+          } else {
+            for (iiii in seq_along(subsubitem)) {
+              subsubsubitem <- subsubitem[[iiii]]
+              if (is.bottom(subsubsubitem)) {
+                args$dataname <- names(x)[i]
+                args$varname <- names(subsubitem)[iiii]
+                args$selected <- subsubsubitem$selected
+                args$keep_na <- subsubsubitem$keep_na
+                args$keep_inf <- subsubsubitem$keep_inf
+                args$datalabel <- names(item)[ii]
+                args$target <- names(subitem)[iiii]
+                slices[[length(slices)+1]] <- as.teal_slice(Filter(Negate(is.null), args))
+                args <- make_args()
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  do.call(filter_settings, slices)
+}
+
+
+# subset method for teal_slices
+#' @export
+#' @rdname new_api
+#' @keywords internal
+#'
+`[.teal_slices` <- function(x, i) {
+  if (missing(i)) i <- seq_along(x)
+  if (length(i) == 0L) return(x[0])
+  if (is.logical(i) & length(i) > length(x)) stop("subscript out of bounds")
+  if (is.numeric(i) & max(i) > length(x)) stop("subscript out of bounds")
+  if (is.character(i) & !all(is.element(i, names(x)))) stop("subscript out of bounds")
+
+  y <- NextMethod("[")
+  attrs <- attributes(x)
+  attrs$names <- attrs$names[i]
+  attributes(y) <- attrs
+  excludes <- unique(unlist(vapply(y, function(ts) ts[["dataname"]], character(1L))))
+  attr(y, "exclude") <- Filter(Negate(is.null), attr(x, "exclude")[excludes])
+  y
+}
+
+
+# concatenate method for teal_slices
+#' @export
+#' @rdname new_api
+#' @keywords internal
+#'
+c.teal_slices <- function(...) {
+  x <- list(...)
+  excludes <- lapply(x, attr, "exclude")
+  names(excludes) <- NULL
+  excludes <- unlist(excludes, recursive = FALSE)
+  excludes <- excludes[!duplicated(names(excludes))]
+  count_types <- lapply(x, attr, "count_type")
+  count_types <- unique(unlist(count_types))
+  checkmate::assert_string(count_types)
+
+  do.call(filter_settings, c(unlist(x, recursive = FALSE), list(exclude = excludes, count_type = count_types)))
+}
+
+
+# print method for teal_slices
+#' @export
+#' @rdname new_api
+#' @keywords internal
+#'
+print.teal_slices <- function(x) {
+  f <- attr(x, "exclude")
+  ct <- attr(x, "count_type")
+  for (i in seq_along(x)) {
+    ind <- names(x)[i]
+    if (is.null(ind)) ind <- sprintf("[[%d]]", i)
+    cat(ind, "\n")
+    print(x[[i]])
+  }
+  cat("\nnon-filterable variables:")
+  if (is.list(f) & length(f) == 0L) {
+    cat(" none\n")
+  } else {
+    cat("\n")
+    for (i in seq_along(f)) {
+      cat(sprintf(" $ %s: %s", names(f)[i], toString(f[[i]])), "\n")
+    }
+  }
+  cat(sprintf("\ncount type: %s", ct), "\n")
+}
+
+
+# get slices where feature is value
+#' @export
+#' @rdname new_api
+#' @keywords internal
```

Comment:
so the export or internal ;] please remove `export` as noted in the comment about NAMESPACE

Reply:
`@keywords internal` merely remove help pages from the index, I don't think S3 methods for generics need to appear there.